cachito/web/models.py

```diff
@@ -597,20 +597,36 @@ class Request(db.Model):
     def __repr__(self):
         return "<Request {0!r}>".format(self.id)
 
-    def add_package(self, package, subpath=None):
+    def add_package(self, package, **association_attrs):
         """
         Associate a package with this request if the association doesn't exist.
 
         Note that the association is added to the database session but not committed.
 
+        Additional attributes to set on the RequestPackage association may be passed via keyword
+        arguments. Currently, only "subpath" is recognized, with the following semantics:
+
+                           | subpath provided           | subpath not provided |
+        |------------------|----------------------------|----------------------|
+        | existing package | check that subpath matches | do nothing           |
+        | new package      | set subpath to value       | set subpath to null  |
+
+        Note that a subpath value of None, "" or "." will be stored as null in the database.
+
         :param Package package: the Package object to associate with the request
-        :param str subpath: custom subpath for package within request repository
+        :param association_attrs: additional attributes to set on the RequestPackage association
         :raises ValidationError: if the association already exists with a different subpath
         """
+        subpath_provided = "subpath" in association_attrs
+
+        subpath = association_attrs.get("subpath")
+        if not subpath or subpath == os.curdir:
```

Comment:
I'm slightly confused about this part `subpath == os.curdir`. The description mentions one condition to set subpath to null which is for new package, when subpath is not provided. Why are we setting it to null when subpath matches current directory?

Reply:
Confusingly, [os.curdir](https://docs.python.org/3/library/os.html#os.curdir) is not actually the current directory, it is the string that the operating system uses to refer to the current directory (`"."` on all the systems that Python supports). Should I use `subpath == "."` instead to make it easier to understand?

Reply:
Oh okay. I think just adding a comment there might suffice

Reply:
+1 to adding a comment there :)